### PR TITLE
fix(profiles): Apply #2330 fix to maintenance branch

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

This PR cherry-picks the fix for #2330 for merging into the maintenance branch.

## Release Notes

Milestone: 2.9.1

Changelog: 

Fixed issue where profiles with authentication tokens were breaking functionality for direct-to-service profiles after user interaction. #2330 

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found

### CODE REVIEW

File: packages/zowe-explorer-api/CHANGELOG.md [hunk 1]
```diff
@@ -18,8 +18,10 @@ All notable changes to the "zowe-explorer-api" extension will be documented in t
 - Added a new type `DataSetAllocTemplate` for data set attributes.
 - Added optional `cancelJob` function to `ZoweExplorerApi.IJes` interface.
 - Added z/OSMF API implementation for `cancelJob` function.
+- Added optional `profile` parameter to `IPromptCredentialsOptions` so developers can choose to skip rebuilding the profile with ProfilesCache.
```
Comment (JillieBeanSim) [inline packages/zowe-explorer-api/CHANGELOG.md:21]: Hey @traeok can we move the changelog items under 2.9.1?  Are they on main under 2.9.0 but didn't go out with that release?  If so we can fix when creating PR from maintenance to main to bring in 2.9.1 changes.

File: packages/zowe-explorer-api/CHANGELOG.md [hunk 2]
```diff
@@ -18,8 +18,10 @@ All notable changes to the "zowe-explorer-api" extension will be documented in t
...
 - Added optional `id` variable to `IZoweTreeNode` interface, which can be used to designate a unique ID for a tree node. [#2215](https://github.com/zowe/vscode-extension-for-zowe/issues/2215)
 - Fixed error shown by API when accessing the `name` and `type` property of a profile when updating the profile arrays [#2334](https://github.com/zowe/vscode-extension-for-zowe/issues/2334).
+- Fixed issue where profiles with authentication tokens were breaking functionality for direct-to-service profiles after user interaction. [#2111](https://github.com/zowe/vscode-extension-for-zowe/issues/2111)
```
Comment (JillieBeanSim) [inline packages/zowe-explorer-api/CHANGELOG.md:24]: dito here on CHANGELOG items

File: packages/zowe-explorer/CHANGELOG.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 - Fixed an issue where the mismatch etag error returned was not triggering the diff editor, resulting in possible loss of data due to the issue. [#2277](https://github.com/zowe/vscode-extension-for-zowe/issues/2277)
 - Fixed issue where refreshing views collapsed the respective trees. [#2215](https://github.com/zowe/vscode-extension-for-zowe/issues/2215)
 - Fixed an issue where user would not get prompted when authentication error is thrown. [#2334](https://github.com/zowe/vscode-extension-for-zowe/issues/2334)
+- Fixed issue where profiles with authentication tokens were breaking functionality for direct-to-service profiles after user interaction. [#2111](https://github.com/zowe/vscode-extension-for-zowe/issues/2111)
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/CHANGELOG.md:30]: dito here on CHANGELOG items

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
Thanks @traeok for bringing these changes over, left some comments on CHANGELOG locations.

File: packages/zowe-explorer-api/CHANGELOG.md (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer-api/CHANGELOG.md:21]: Yes, sorry for the confusion - will update. I think it was added to the changelog under 2.9 at the time it was marked as ready, but was merged after the release.

Comment (JillieBeanSim) [reply packages/zowe-explorer-api/CHANGELOG.md]: no worries, thanks for updating.  The same situation slipped by me last time till it was pointed out in release PR (thank goodness 😆 )

**Review by JillieBeanSim — DISMISSED** [review]

**Review by JillieBeanSim — DISMISSED** [review]

**Review by JillieBeanSim — DISMISSED** [review]

**Review by JillieBeanSim — APPROVED** [review]

**Review by rudyflores — APPROVED** [review]

**Pull request merged**